addons/delivery/models/stock_picking.py

```diff
@@ -115,10 +116,15 @@ def _compute_carrier_tracking_url(self):
     @api.depends('carrier_id', 'move_ids_without_package')
     def _compute_return_picking(self):
         for picking in self:
-            if picking.carrier_id and hasattr(picking.carrier_id, '%s_get_return_label' % picking.carrier_id.delivery_type):
+            if picking.carrier_id and picking.carrier_id.can_generate_return:
                 picking.is_return_picking = any(m.origin_returned_move_id for m in picking.move_ids_without_package)
             else:
                 picking.is_return_picking = False
+
+    def _compute_return_label(self):
+        for picking in self:
+            picking.return_label_ids = self.env['ir.attachment'].search([('res_model', '=', 'stock.picking'), ('res_id', '=', picking.id), ('name', 'ilike', 'ReturnLabel')])
```

Comment:
'ReturnLabel' + message_post in generic _get_return_label

Reply:
ilike and ReturnLabel is weird to me, ilike is case insensitive
can't you prepend a fixed word to these return label? this would make the lookup a little more robust (=like fixeprefix_%